### DISCUSSION TITLE
mesh.mark_entities() vs RelabeledMesh()

### DISCUSSION THREAD

**Discussion by miguelcoolchips** [firedrakeproject/firedrake#3126] [Firedrake support] [answered]
I want to mark cells in a mesh directly with `mark_entities()` with the following script:
```python
import firedrake as fd
from firedrake import conditional, And, Or
from firedrake.cython.dmcommon import CELL_SETS_LABEL


def test_mark_entities_mesh_mark_entities_2d():
    label_name = "test_label"
    label_value = 999
    mesh = fd.UnitSquareMesh(40, 8)
    x, y = fd.SpatialCoordinate(mesh)
    V = fd.FunctionSpace(mesh, "DP", 0)
    f = fd.Function(V).interpolate(conditional(x < 0.25, 1., 0.))
    mesh.mark_entities(f, label_name, label_value)
    #mesh = fd.RelabeledMesh(mesh,
    #                 [f],
    #                 [label_value])
    plex = mesh.topology.topology_dm
    value = fd.assemble(fd.Constant(1.0) * fd.dx(label_value, domain=mesh))
    print(value)

if __name__ == "__main__":
    test_mark_entities_mesh_mark_entities_2d()

```
but the result of `value` is 0, i.e. no marking happening. If I uncomment the call to `RelabeledMesh`, I get the correct value (0.25). I discovered that it might be due to the fact that we are marking the mesh with a different label, `label_name` and not with the label used in `RelabeledMesh`, `CELL_SETS_LABEL`. However, I cannot use the latter in `mark_entities` (exception thrown). I'd rather avoid using `RelabeledMesh`. What are my options?

**Comment by ksagiyam** [comment, marked as answer]
You are right. This PR https://github.com/firedrakeproject/firedrake/pull/3128 allows for using `label_name=CELL_SETS_LABEL` and `label_name=FACE_SETS_LABEL` and also make them defaults.
Your example will then work by replacing:
```
mesh.mark_entities(f, label_name, label_value)
```
with
```
mesh.mark_entities(f, label_value)
```
.

**Reply by miguelcoolchips** [reply]
Thanks!
